sdk/core/core-amqp/src/amqpAnnotatedMessage.ts

```diff
@@ -62,12 +63,30 @@ export const AmqpAnnotatedMessage = {
       properties: AmqpMessageProperties.fromRheaMessageProperties(msg),
       body: msg.body,
     };
+    if (msg.absolute_expiry_time) {
+      const absoluteExpiryTime = msg.absolute_expiry_time.getTime();
+      amqpMsg.properties.absoluteExpiryTime =
+        absoluteExpiryTime >= Constants.maxAbsoluteExpiryTime
+          ? Constants.maxAbsoluteExpiryTime
+          : absoluteExpiryTime;
+      // The TTL from the header can be at most approximately 49 days (uint32
+      // max value milliseconds) due to the AMQP spec. In order to allow for
+      // larger TTLs set by the user, we take the difference of the
+      // absolute_expiry_time and the creation_time (if both are set). If either of
+      // those properties is not set, we fall back to the TTL from the header.
```

Comment:
When would creation_time be not set?

Reply:
possibly for messagse created from other clients?